[DO-518] How to send payment with Gevamu (#4)
gevamu/gevamu.github.io: content/4.examples/send-payment.draft.md

  }

  MyPaymentService ..> GevamuFacade
```

## Program implementation

Sending payment is quite simple operation. It includes only call of `PaymentFlow`. 

But before starting the flow you should provide payment instruction. It should be created according to chosen payment standard.

```kotlin
class MyPaymentService(private val gevamuFacade: GevamuFacade) {
    fun sendPaymentViaGevamu() {
        // Create payment instruction with payment details
        val paymentInstruction = PaymentInstruction(/* payment details */)
        
        // Specify gateway node for payment processing
        val gateway = Party(/* gateway node data */)
        
        // Use GevamuFacade to send payment instruction to gateway node
        gevamuFacade.sendPayment(paymentInstruction, gateway)
    }
}

class GevamuFacade(private val serviceHub: AppServiceHub): SingletonSerializeAsToken() {

  fun sendPayment(
    paymentInstruction: PaymentInstruction,
    gateway: Party
  ): List<StateAndRef<Payment>> {
    val flowHandle = serviceHub.startFlow(PaymentFlow(paymentInstruction, gateway, UUID.randomUUID()))
    return flowHandle.returnValue.get()
  }
}
```

## Transactions lifecycle

> Once payment instruction is received, the Gevamu Payment Solution makes an authentication (using node identity and the Participant ID).

Reviewer: ```suggestion
Once payment instruction is received, the Gevamu Payment Solution validates and authorizes (by checking node identity and the Participant ID) the payment.
```

> If the validation passes, the Payment Gateway returns the ‘Sent to Gateway’ status back to the Participant.

Reviewer: ```suggestion
If the validation passes, the Payment Gateway sets the payment status to ‘Sent to Gateway’. Status change is communicated back to the Participant's node.
```

Transaction can be accepted or rejected by the Payment Service Provider.

To make a transaction as a network participant, describe Payment in one of formats listed in [`PaymentInstructionFormat`](https://gevamu.github.io/corda-payments-sdk/payments-workflows/com.gevamu.corda.flows/-payment-instruction-format/index.html) enum.

You can find sequence diagrams of payment processing and payment state transition below:

```mermaid

sequenceDiagram
  participant CA as MyPaymentService
  participant PCD as GevamuFacade

  CA->>PCD: Initiate Payment
  PCD->>PCD: Initiate payment <br/> within PSP
  PCD->>CA: Payment status: Created
  PCD->>PCD: Passive payment update...
  CA->>PCD: Query payment

>   PCD->>CA: Payment with relevant status

Reviewer: I'm not sure it's a part of the flow.

Reviewer 2: It is not a part of flow, but it is what user should do in order to get relevant status. @StanG4313 perhaps we need to separate this part of diagram somehow

Reviewer: It's possible to subscribe to vault updates with `trackBy()` https://docs.r3.com/en/platform/corda/4.9/enterprise/cordapps/api-vault-query.html

I think there was also a possibility to start a flow when corda state changes, but I cannot quickly find it.

Author: Fixed in new commit

```

```mermaid

> stateDiagram-v2

Reviewer: There is probably the Pending status. I need to check it.

Reviewer 2: @ASitnikov pending status [exists](https://gevamu.github.io/corda-payments-sdk/payments-contracts/com.gevamu.corda.states/-payment/-payment-status/index.html), but as I remember, it exists only inside Gateway and Participant can't see it.

Reviewer: It's visible to the payer (participant)

  }

  MyPaymentService ..> GevamuFacade
```

## Program implementation

Sending payment is quite simple operation. It includes only call of `PaymentFlow`. 

But before starting the flow you should provide payment instruction. It should be created according to chosen payment standard.

```kotlin
class MyPaymentService(private val gevamuFacade: GevamuFacade) {
    fun sendPaymentViaGevamu() {
        // Create payment instruction with payment details
        val paymentInstruction = PaymentInstruction(/* payment details */)
        
        // Specify gateway node for payment processing
        val gateway = Party(/* gateway node data */)
        
        // Use GevamuFacade to send payment instruction to gateway node
        gevamuFacade.sendPayment(paymentInstruction, gateway)
    }
}

class GevamuFacade(private val serviceHub: AppServiceHub): SingletonSerializeAsToken() {

  fun sendPayment(
    paymentInstruction: PaymentInstruction,
    gateway: Party
  ): List<StateAndRef<Payment>> {
    val flowHandle = serviceHub.startFlow(PaymentFlow(paymentInstruction, gateway, UUID.randomUUID()))
    return flowHandle.returnValue.get()
  }
}
```

> To make a transaction as a network participant, describe Payment in one of formats listed in [`PaymentInstructionFormat`](https://gevamu.github.io/corda-payments-sdk/payments-workflows/com.gevamu.corda.flows/-payment-instruction-format/index.html) enum. 

Reviewer 2: This sentence can easily continue paragraph before diagram. Let's move it.

Author: Did it in new commit
